ibis/common/collections.py

```diff
@@ -287,6 +287,11 @@ def __init__(self, *args, **kwargs):
     def __hash__(self) -> int:
         return self.__precomputed_hash__
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, collections.abc.Mapping):
+            return NotImplemented
+        return collections.OrderedDict(self) == collections.OrderedDict(other)
```

Comment:
I assume the point of this change is to turn differently ordered projections unequal.

Since we use `FrozenDict` heavily across the codebase how about adding a `FrozenOrderedDict(FrozenDict)` for this purpose using `self.items()` for comparison?

Then `FrozenOrderedDict` could be used for the relevant relational operations instead of `FrozenDict`.

Reply:
Agreed, let's do that after the 9.0 release!